[FF-7] Migrate Tabs

dmaklygin opened the pull request:
[FF-7]

### Description

Migrate tabs to Mui/Base

### How to test

<!-- The temploy link will be automatically updated when the temploy is deployed -->
- [Temploy](https://picasso.toptal.net/ff-7-tabs)
- FIXME: Add the steps describing how to verify your changes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Development checks

- [ ] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [ ] Double check if `picasso-tailwind-merge` requires major update (check its `README.md`)
- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [ ] Self reviewed
- [ ] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- [ ] codemod is created and showcased in the changeset
- [ ] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping for reviews

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FF-7]: https://toptal-core.atlassian.net/browse/FF-7?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dmaklygin commented:
@toptal-bot run package:alpha-release

dmaklygin commented:
@toptal-bot run package:alpha-release

dmaklygin commented:
@toptal-bot run package:alpha-release

george-aidonidis commented:
@dmaklygin I know this is a draft but https://github.com/toptal/staff-portal/pull/14453 you requested to review this too. Seems in this story the active tab is not working anymore 
<img width="578" alt="52282" src="https://github.com/user-attachments/assets/7ae3dae0-63c2-45d3-886a-d6db66f40b9d" />

dmaklygin commented:
> @dmaklygin I know this is a draft but [toptal/staff-portal#14453](https://github.com/toptal/staff-portal/pull/14453) you requested to review this too. Seems in this story the active tab is not working anymore <img alt="52282" width="578" src="https://private-user-images.githubusercontent.com/11461413/453264049-7ae3dae0-63c2-45d3-886a-d6db66f40b9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDk1NDc2NjgsIm5iZiI6MTc0OTU0NzM2OCwicGF0aCI6Ii8xMTQ2MTQxMy80NTMyNjQwNDktN2FlM2RhZTAtNjNjMi00NWQzLTg4NmEtZDZkYjY2ZjQwYjlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjEwVDA5MjI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2ZDljMzBhZjA3NjY4ZGI0NjAxM2VhYjJiZGI1YmU3NjYyMjY5OTQ2YWUxNTA3M2Q1ZTQ5NDhhN2M5MjljZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.v2NKnnkgfIV6qN2y0j6AQ4TrNYY_fav7eAzKMzrT07Y">

Could you please share a link to temploy. 
I can check this temploy https://picasso.toptal.net/ff-7-tabs/?path=/story/layout-tabs--tabs and everything look working

dmaklygin commented:
> @dmaklygin but the story in this PR is different please check the link and notice the change on https://happo.io/a/675/p/1189/compare/ea8558d664d3d3c12c47263852996790aed70a84/2a3da7cea3ba840b0a702d9efc91380d60ee2fe6#Tabs%5C.Tab-Using%5C%20custom%5C%20value

Yeah, I've checked the stories and confirm, the changes are intentional.
Since, I've changed the default value for the `Tabs.Tab Using custom value`
Also, fixed the icons for the `Tabs.Tab With Icon or Badge`

george-aidonidis commented:
My apologies for the confusion 🙈
